packages/nouns-contracts/contracts/governance/NounsDAOLogicV3.sol

```diff
@@ -419,6 +424,59 @@ contract NounsDAOLogicV3 is NounsDAOStorageV2, NounsDAOEventsV2 {
         emit ProposalVetoed(proposalId);
     }
 
+    function ragequit(uint256[] calldata nounIds) external {
+        uint256 supply = nounsCirculatingSupply();
+
+        // transfer all nouns to timelock
+        // will revert if not owner or has approval for all nouns
+        // TODO: should this be allowed only for owner of the nouns?
+        for (uint256 i = 0; i < nounIds.length; ++i) {
+            nouns.transferFrom(msg.sender, address(timelock), nounIds[i]);
+        }
+
+        ////// ETH redeem
+
+        // calculate ETH pro rata share
+        uint256 proRataShare = (nounIds.length * address(timelock).balance) / supply;
+
+        // apply penalty
+        uint256 amountOfEthToRedeem = ((10_000 - ragequitPenaltyBPs) * proRataShare) / 10_000;
+
+        // send ETH to msg.sender
+        timelock.redeem(msg.sender, amountOfEthToRedeem);
+
+        ////// ERC20s redeem
+        for (uint256 i = 0; i < redeemableAssets.length; ++i) {
+            address asset = redeemableAssets[i];
+
+            proRataShare = (nounIds.length * IERC20(asset).balanceOf(address(timelock))) / supply;
+            uint256 amountOfTokensToRedeem = ((10_000 - ragequitPenaltyBPs) * proRataShare) / 10_000;
+            IERC20(asset).transferFrom(address(timelock), msg.sender, amountOfTokensToRedeem);
+        }
+    }
+
+    function _setRagequitPenaltyBPs(uint32 ragequitPenaltyBPs_) external {
+        if (msg.sender != admin) {
+            revert AdminOnly();
+        }
+
+        // TODO: check max bps value?
+
+        ragequitPenaltyBPs = ragequitPenaltyBPs_;
+    }
+
+    function _addRedeemableAsset(address erc20) external {
```

Comment:
as we discussed, we want to explore the approach of letting all treasury-owned ERC20s be redeemable by default
to make ragequit fair by default without having to get DAO permission and attention to whitelist tokens via proposals